examplemusig2.c

```diff
@@ -49,34 +50,32 @@ int main(void) {
         /* Store the public key of the signer in pk_list */
         assert (secp256k1_keypair_pub(ctx, &pk_list[i], &mcs_list[i].keypair));
 
-        /* Store the batch commitments of the signer in batch_list */
+        /* Store the batch commitments of the signer in serialized batch_list */
         l = 0; // the index of the signer's commitment list.
         for (k = 0; k < NR_MESSAGES; k++) {
             for (j = 0; j < V; j++, l++) {
-                assert(secp256k1_keypair_pub(ctx, &batch_list[k][i][j], mcs_list[i].comm_list[l]));
+                ind = (k * NR_SIGNERS * V + i * V + j) * SER_PK_BYTES;
+                assert(secp256k1_keypair_pub(ctx, &temp_comm, mcs_list[i].comm_list[l]));
+                secp256k1_ec_pubkey_serialize(ctx, &serialized_batch_list[ind], &ser_size, &temp_comm, SECP256K1_EC_UNCOMPRESSED );
             }
         }
     }
     printf("--------------------------------------------------------------------------- \n\n");
 
-
-
-    printf("**** STATE 1 ************************************************************** \n");
-
-    // todo: idea: would  be good to 'prepare' signers for all the batch. This prepare
-    // function would compute the aggr pk, and all agr R values.
-    /**** Aggregate the public keys and batch commitments for each signer ****/
+    /**** Aggregate the public keys and batch commitments for each signer for all messages ****/
     int cnt = 0;
     for (i = 0; i < NR_SIGNERS; i++)
-        cnt += musig2_aggregate_pubkey(&mcs_list[i].mc, pk_list, NR_SIGNERS);
-    printf("* %d signers aggregated public key.\n", cnt);
+        cnt += musig2_signer_precomputation(&mcs_list[i].mc, pk_list, serialized_batch_list, NR_SIGNERS, NR_MESSAGES);
 
-    cnt = 0;
-    for (i = 0; i < NR_SIGNERS; i++)
-        cnt += musig2_aggregate_R(&mcs_list[i].mc, batch_list[mcs_list[i].state]);
-    printf("* %d signers aggregated commitment.\n", cnt);
+    // todo: Do we need to free all signers?
+    if (cnt != NR_SIGNERS){
+        for (k = 0; k < NR_SIGNERS; k++)
+            musig2_context_sig_free(&mcs_list[k]);
+        return -1;
+    }
```

Comment:
If you fail now, then you probably want to do it earlier when the signers are initialised. And if you are going to return -1, then there is no sense on having `cnt` (same applies to above). If any call fails, then the whole example fails. 

Then, you only need to free up to the index you've reached successfully.